### PR TITLE
[chain id] rm panic in ChainId from_str

### PR DESCRIPTION
**Pull request by christinacelee**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/libra/developers.libra.org, and link to your PR here.)

### CODE REVIEW

File: types/src/chain_id.rs [hunk 1]
```diff
@@ -138,8 +138,12 @@ impl FromStr for ChainId {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self> {
-        assert!(!s.is_empty());
-        NamedChain::str_to_chain_id(s).or_else(|_err| Ok(ChainId::new(s.parse::<u8>()?)))
+        ensure!(!s.is_empty(), "Cannot create chain ID from empty string");
```
Comment (thefallentree) [inline types/src/chain_id.rs:141]: please add a testcase to make sure it won't come back.

**Review by davidiw — APPROVED** [review]
do we really need the ensure statement? and what happens if I give you 255255, does that parse to a u8 or does it error?

**Comment by davidiw** [comment]
Was this causing any issues?

**Pull request closed without merging**